core/modules/image/image.field.inc

```diff
@@ -542,9 +556,54 @@ function image_field_widget_process($element, &$form_state, $form) {
     '#access' => (bool) $item['fid'] && $settings['title_field'],
   );
 
+  // Pass alt and title required status to JS in case other modules need access.
+  $element['#attached']['js'][] = array(
+    'type' => 'setting',
+    'data' => array(
+      'image' => array(
+        'altRequired' => $instance['settings']['alt_field_required'],
+      ),
+    ),
+  );
+
+  if ($settings['alt_field_required']) {
+    $element['alt']['#title'] .= ' <span class="form-required" title="' . t('The alt text field is required.') . '">*</span>';
```

Comment:
Manually appending to the title seems like it could cause some problems in some theming situations. Also, I think this markup doesn't match other required attributes, which use `<abbr>` instead of `<span>`.

Rather than manual markup, we could try adding an element pre_render and add the `#required` property there, so it renders but doesn't actually provide its own validation.
```suggestion
    $element['alt']['#pre_render'] = array('_image_field_widget_alt_pre_render`);
```

Reply:
Fixed with [`7647357` (#4422)](https://github.com/backdrop/backdrop/pull/4422/commits/7647357614d5b91f84f34312a417ba1118344b60)